core/src/test/scala/org/apache/spark/serializer/JavaSerializerSuite.scala

```diff
@@ -25,4 +25,18 @@ class JavaSerializerSuite extends SparkFunSuite {
     val instance = serializer.newInstance()
     instance.deserialize[JavaSerializer](instance.serialize(serializer))
   }
+
+  test("Deserialize object containing a primitive Class as attribute") {
+    val serializer = new JavaSerializer(new SparkConf())
+    val instance = serializer.newInstance()
+    instance.deserialize[JavaSerializer](instance.serialize(new ContainsPrimitiveClass()))
+  }
```

Comment:
It would require the other classloader to not have access to that class (so wouldn't work with the existing classes). The only way I see would be to generate some class at runtime (via asm or whatever) and make it available only to the classloader used during serialization. But that sounds kind of overkill to me.